x-pack/plugins/reporting/server/export_types/printable_pdf/printable_pdf.ts

```diff
@@ -0,0 +1,130 @@
+    LICENSE_TYPE_ENTERPRISE,
+  ];
+
+  constructor(...args: ConstructorParameters<typeof ExportType>) {
+    super(...args);
+    const logger = args[2];
+    this.logger = logger.get('png-export-v1');
+  }
+
+  public createJob = async (
+    { relativeUrls, ...jobParams }: JobParamsPDFDeprecated // relativeUrls does not belong in the payload of PDFV1
+  ) => {
+    validateUrls(relativeUrls);
+
+    // return the payload
+    return {
+      ...jobParams,
+      isDeprecated: true,
+      forceNow: new Date().toISOString(),
+      objects: relativeUrls.map((u) => ({ relativeUrl: u })),
+    };
+  };
+
+  public runTask = async (
+    jobId: string,
+    job: TaskPayloadPDF,
+    cancellationToken: CancellationToken,
+    stream: Writable
+  ) => {
+    const jobLogger = this.logger.get(`execute-job:${jobId}`);
+    const apmTrans = apm.startTransaction('execute-job-pdf', REPORTING_TRANSACTION_TYPE);
+    const apmGetAssets = apmTrans?.startSpan('get-assets', 'setup');
+    let apmGeneratePdf: { end: () => void } | null | undefined;
+
+    const process$: Rx.Observable<TaskRunResult> = Rx.of(1).pipe(
+      Rx.mergeMap(() => decryptJobHeaders(this.config.encryptionKey, job.headers, jobLogger)),
+      Rx.mergeMap(async (headers: any) => {
+        const fakeRequest = this.getFakeRequest(headers, job.spaceId, jobLogger);
+        const uiSettingsClient = await this.getUiSettingsClient(fakeRequest);
+        return getCustomLogo(uiSettingsClient, headers);
+      }),
+      Rx.mergeMap(({ headers, logo }) => {
+        // @ts-ignore relativeUrl issue from deprecated export type
+        const urls = getFullUrls(this.getServerInfo(), this.config, job);
+
+        const { browserTimezone, layout, title } = job;
+        apmGetAssets?.end();
+
+        apmGeneratePdf = apmTrans?.startSpan('generate-pdf-pipeline', 'execute');
+        //  make a new function that will call reporting.getScreenshots
+        const snapshotFn = () =>
+          this.startDeps.reporting.getScreenshots({
+            format: 'pdf',
+            title,
+            logo,
+            urls,
+            browserTimezone,
+            headers,
+            layout,
+          });
+        return generatePdfObservable(snapshotFn, {
+          format: 'pdf',
+          title,
+          logo,
+          urls,
+          browserTimezone,
+          headers,
+          layout,
+        });
+      }),
+      Rx.tap(({ buffer }) => {
+        apmGeneratePdf?.end();
+        if (buffer) {
+          stream.write(buffer);
+        }
+      }),
+      Rx.map(({ metrics, warnings }) => ({
+        content_type: 'application/pdf',
+        metrics: { pdf: metrics },
+        warnings,
+      })),
+      Rx.catchError((err: any) => {
+        jobLogger.error(err);
+        return Rx.throwError(err);
+      })
+    );
+
+    const stop$ = Rx.fromEventPattern(cancellationToken.on);
+
+    apmTrans?.end();
+    return Rx.lastValueFrom(process$.pipe(Rx.takeUntil(stop$)));
```

Comment:
So that we can keep the imports the same:
```suggestion
    return Rx.lastValueFrom(process$.pipe(takeUntil(stop$)));
```